File: Go task/code.go
package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
)

// Order represents the structure of an order
type Order struct {
	ID       int    `json:"id"`
	Item     string `json:"item"`
	Quantity int    `json:"quantity"`
	State    string `json:"state"`
	// TODO: Add more fields here (like order_date, processed_by)
}

func main() {
	// Step 1: Read new_orders.json
	data, err := ioutil.ReadFile("new_orders.json")
	if err != nil {
		fmt.Println("Error reading file:", err)
		return
	}

	var orders []Order
	err = json.Unmarshal(data, &orders)
	if err != nil {
		fmt.Println("Error parsing JSON:", err)
		return
	}

	// Step 2: Process each order
	for i := range orders {
		// TODO: Transform each order
		// - Change state from "new" to "picked"
		// - Add order_date (today's date)
		// - Add processed_by ("intern_name")
		// - Bonus: If item is "Massage Oil", increase quantity by 10%
	}

	// Step 3: Write to processed_orders.json
	output, err := json.MarshalIndent(orders, "", "  ")
	if err != nil {
		fmt.Println("Error encoding JSON:", err)
		return
	}

	err = ioutil.WriteFile("processed_orders.json", output, 0644)
	if err != nil {
		fmt.Println("Error writing file:", err)
		return
	}

	fmt.Println("Orders processed successfully!")
}

File: Python task/code.py
import json
from datetime import date

# Read new orders from JSON file
with open('new_orders.json', 'r') as file:
    new_orders = json.load(file)

# Process each order
processed_orders = []
for order in new_orders:
    order['state'] = 'picked'  # Transform state
    order['order_date'] = str(date.today())  # Add today's date
    processed_orders.append(order)

# Write processed orders to another JSON file
with open('processed_orders.json', 'w') as file:
    json.dump(processed_orders, file, indent=2)

print("Orders processed successfully!")